### PR TITLE
Fix parameterized test support (spoon-runner)

### PR DESCRIPTION
**Pull request by mabae**
This PR addresses issue #533

### CODE REVIEW

File: spoon-runner/src/main/java/com/squareup/spoon/LogRecordingTestRunListener.kt [hunk 1]
```diff
@@ -1,15 +1,26 @@
 package com.squareup.spoon
 
+import com.android.annotations.VisibleForTesting
 import com.android.ddmlib.testrunner.ITestRunListener
 import com.android.ddmlib.testrunner.TestIdentifier
 import com.google.common.collect.ImmutableList
-import java.util.LinkedHashSet
+import java.util.*
```
Comment (edenman) [inline spoon-runner/src/main/java/com/squareup/spoon/LogRecordingTestRunListener.kt]: no star imports

File: spoon-runner/src/test/java/com/squareup/spoon/LogRecordingTestRunListenerTest.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
+package com.squareup.spoon
+
+import com.android.ddmlib.testrunner.TestIdentifier
+import com.squareup.spoon.LogRecordingTestRunListener.Companion.stripParametersInClassName
+import org.hamcrest.CoreMatchers.*
+import org.junit.Assert.assertThat
```
Comment (edenman) [inline spoon-runner/src/test/java/com/squareup/spoon/LogRecordingTestRunListenerTest.kt]: let's switch this to use Truth instead (see `DeviceUtilsTest` for example).  Side benefit of consistency with the other tests: no nasty backticks around the names of hamcrest matchers.

**Comment by mabae** [comment]
Thanks for the feedback. I updated the PR accordingly.

**Comment by edenman** [comment]
@mabae one more thing: the test should be 2-space indented, not 4.

**Comment by edenman** [comment]
Thanks @mabae!

**Pull request merged**